assets/js/script.js

```diff
@@ -0,0 +1,164 @@
+        <li class="custom" >UV Index: <span id="uvi"></span></li>
+      </ul>
+  </div>`;
+};
+
+const renderTodaysWeather = function (data) {
+  currentWeather.empty();
+  const topBlock = constructTodaysWeather(data);
+  currentWeather.append(topBlock);
+};
+
+const constructForecast = function (dailyArray) {
+  dailyArray.forEach((element) => {
+    const date = moment.unix(element.dt).format("DD-MM-YYYY");
+    forecastBlock = `<div class="card col-2 padstyle" style="width: 18rem;">
+    <div class="card-body">
+      <h5 class="card-title">${date}<img src="http://openweathermap.org/img/wn/${element.weather[0].icon}.png" alt="..."></h5>
+    </div>
+    <ul class="list-group list-group-flush">
+      <li class="list-group-item">Temperature: ${element.temp.day} <sup>O</sup>C</li>
+      <li class="list-group-item">Humidity: ${element.humidity} %</li>
+      <li class="list-group-item">Wind speed: ${element.wind_speed} m/s</li>
+    </ul>
+</div>`;
+    forecastContainer.append(forecastBlock);
+  });
+};
+
+const renderForecast = function (dailyArray) {
+  forecastContainer.empty();
+  constructForecast(dailyArray);
+};
+
+const getFromLocalStorage = function () {
+  const searchHistory = JSON.parse(localStorage.getItem("history")) || [];
+  return searchHistory;
+};
+
+const handleResponse = function (response) {
+  return response.json();
+};
+
+const showError = function () {
+  const error = document.createElement("p");
+  error.textContent = "please enter a valid city";
+  $("#error-response").append(error);
+};
+
+const renderUviSpan = function (uviData) {
+  const placeHolder = $("#uvi");
+  placeHolder.text(uviData);
+  if (uviData < 3) {
+    placeHolder.addClass("green");
+  } else if (uviData >= 3 && uviData < 8) {
+    placeHolder.addClass("orange");
+  } else {
+    placeHolder.addClass("red");
+  }
+};
+
+const renderPage = function (city) {
+  const myUrl =
+    `https://api.openweathermap.org/data/2.5/weather?q=${city}&units=metric&APPID=` +
+    myAPIKey;
+
+  const handleData = function (data) {
+    if (data.cod == 200) {
+      //   save city into local storage
+      const searched = getFromLocalStorage();
+      if (!searched.includes(data.name)) {
+        searched.push(data.name);
+        localStorage.setItem("history", JSON.stringify(searched));
+      }
+
+      renderSearchHistory();
+
+      renderTodaysWeather(data);
+
+      // get data for forecast fetch
+
+      const forecastURL = `https://api.openweathermap.org/data/2.5/onecall?lat=${data.coord.lat}&lon=${data.coord.lon}&exclude={part}&units=metric&appid=${myAPIKey}`;
+
+      const handleSecondData = function (data) {
+        const uviData = data.current.uvi;
+        const dailyArray = data.daily;
+        const dailyArr = dailyArray.slice(1, 6);
+
+        renderForecast(dailyArr);
+        renderUviSpan(uviData);
+      };
+
+      fetch(forecastURL).then(handleResponse).then(handleSecondData);
```

Comment:
Would be nice to see you using `async await` rather than `.then` - as your code gets longer, the `async await` will be needed and will keep your code more readable

Reply:
feel like after project I've got my head around this concept